File: Pong Invaders/Assets/Scripts/Player/playerProjectile.js
#pragma strict

var speed = 0.2;
var direction = 1;
var posX;
var posY;

function Start () {
	
}

function Update () {
	// Position = position + speed
	gameObject.transform.position.y += speed * direction;
	
	/*
	// shorthand variables for position
	posX = gameObject.transform.position.x;
	posY = gameObject.transform.position.y;
	
	
	// Destroys projectiles once they're out of camera range
	if (posX < 0 || posX > pixelWidth){
		Destroy(gameObject);
	}
	else if (posY < 0 || posY > pixelHeight){
		Destroy(gameObject);
	}
	*/
}

File: Pong Invaders/Assets/Scripts/Player/playerProjectile.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

//handles the actual projectile for player 1
public class playerProjectile : MonoBehaviour 
{
	
	public float speed = 0.5f;
	int direction = 1;
	Vector3 velocity = new Vector3(0,0,0);
	
	// Use this for initialization
	void Start () 
	{
		
	}
	
	// Update is called once per frame
	void Update ()
	{
		// Position = position + speed
		velocity.y = speed * direction;
		gameObject.transform.position += velocity;
	}

	//destroys projectile upon impact
	void OnTriggerEnter2D(Collider2D coll) 
	{
        if(coll.gameObject.tag!= "Player1" && coll.gameObject.name!="blueDrone")
		{
		Destroy (gameObject);
        }
	}

	//destroys projectile upon impact
	void OnCollisionEnter2D(Collision2D coll) 
	{
		if (coll.gameObject.tag == "Ball")
		{
			Destroy (gameObject);
		}
	}
}
